metricbeat/mb/mb.go

```diff
@@ -306,17 +308,49 @@ type ModuleConfig struct {
 	MetricSets []string      `config:"metricsets"`
 	Enabled    bool          `config:"enabled"`
 	Raw        bool          `config:"raw"`
+	Query      QueryParams   `config:"query"`
 }
 
 func (c ModuleConfig) String() string {
 	return fmt.Sprintf(`{Module:"%v", MetricSets:%v, Enabled:%v, `+
-		`Hosts:[%v hosts], Period:"%v", Timeout:"%v", Raw:%v}`,
+		`Hosts:[%v hosts], Period:"%v", Timeout:"%v", Raw:%v, Query:%v}`,
 		c.Module, c.MetricSets, c.Enabled, len(c.Hosts), c.Period, c.Timeout,
-		c.Raw)
+		c.Raw, c.Query)
 }
 
 func (c ModuleConfig) GoString() string { return c.String() }
 
+// QueryParams is a convenient map[string]interface{} wrapper to implement the String interface which returns the
+// values in common query params format (key=value&key2=value2) which is the way that the url package expects this
+// params (without the initial '?')
+type QueryParams map[string]interface{}
+
+// String returns the values in common query params format (key=value&key2=value2) which is the way that the url
+// package expects this params (without the initial '?')
+func (q QueryParams) String() (s string) {
+	u := url.Values{}
+
+	for k, v := range q {
+		if values, ok := v.([]interface{}); ok {
+			for _, innerValue := range values {
+				u.Add(k, fmt.Sprintf("%v", innerValue))
+			}
+		} else {
+			//nil values in YAML shouldn't be stringified anyhow
+			if v == nil {
+				u.Add(k, "")
+			} else {
+				u.Add(k, fmt.Sprintf("%v", v))
```

Comment:
Could you explain why we need to stringify? I would expect this to be a string already

Reply:
Actually, `v` comes from `queryParams` which is the `map[string]interface{}` that is used to parse the incoming data from the YAML. That's why it should be stringified :)

Reply:
Should it be considered a configuration error if `v` is not a string here? I see the list use case is handled before

Reply:
I think I don't understand the question, but maybe I'm missing something. You can pass to string pretty much everything with `%v`, be it a `byte` a `rune` or a `float64` it will be stringified to `string` anyways with `fmt.Sprintf("%v")`

Reply:
ok, I'm good with it